### ISSUE TITLE
播放9路视频失败，其中个别会失败，但不是必定，失败报错如下

### ISSUE THREAD

**Issue by tshug** [Bilibili/flv.js#126]
Uncaught (in promise) DOMException: Failed to load because no supported source was found.

**Comment by xqq** [comment]
浏览器不是chrome? 视频参数？

**Comment by tshug** [comment]
是chrome，有720P的，也有1080P的



--

 顺颂商祺！ 
四川华雁信息产业股份有限公司
唐曙光
2017年4月14日
------------------------------------------ 
 

在 2017-04-14 16:57:49，"xqq" <notifications@github.com> 写道：


浏览器不是chrome? 视频参数？

**Comment by xqq** [comment]
给console和media-internals日志

**Comment by xqq** [comment]
@baxerus 
It may caused by the `HE-AAC` audio codec (mp4a.40.5), which was handled by slightly different code between `flv.js` and `hls.js`.

It's better if you can provide reproduceable sample, just use `curl/wget` to pull part of the stream to file.

**Comment by baxerus** [comment]
Yeah, I have seen somewhere in the `hls.js` logging (or code, I actually can not find it again in the moment) that `HE-AAC` and `AAC` are defined at the same time. And `hls.js` is choosing `AAC`.

I will capture a sample now. Give me some minutes

**Comment by xqq** [comment]
The AAC header processing code is in
https://github.com/Bilibili/flv.js/blob/master/src/demux/flv-demuxer.js#L631

**Comment by baxerus** [comment]
I recorded about 1 minute of the RTMP stream with `FFmpeg` (and codec copy). You can find it [here](http://baxerus.com/no-auth/example.flv) (about 75Mb).

**Comment by baxerus** [comment]
I poked around in the `flv-demuxer`, `mp4-remuxer` and other code. But till now, no hacking worked. Most likely it is simply over my head what is happening there. I will try some more, but I think I will not be able to get it to run...

**Comment by xqq** [comment]
The sample video works well without any problem on my chrome, also works well in FireFox.

What's your operating system and platform?

![image](https://user-images.githubusercontent.com/4645762/44916501-dec20600-ad70-11e8-991d-3ecc824637bf.png)

**Comment by baxerus** [comment]
Okay, that is the right picture 😉

**Comment by baxerus** [comment]
I wasn't precise in the description how we use it, sorry.

When I place the `example.flv` file on a webserver and try to play it, it is also working for me. But when it is played over a WSS connection created by `https://github.com/illuspas/Node-Media-Server` it isn't working anymore (but other streams are working. Nearly all).

I try playing it back with `Chrome 68.0.3440.106` on `Mac OS X 10.13.4`

**Comment by xqq** [comment]
Is it works over `http-flv` stream served by the `Node-Media-Server`?

**Comment by baxerus** [comment]
Actually I didn't tried `flv over http` with `Node-Media-Server`, because I can't use it without WSS because of other reasons.

Should I try this?

**Comment by xqq** [comment]
If it has difficulty with `Mixed Content` problem (http request inside a https page),
place a `nginx` reserve proxy in front of the `Node-Media-Server` can be helpful to enable https for http-flv.

**Comment by baxerus** [comment]
No, it is not about `https`. There is a `nginx` before the `Node-Media-Server` doing Websockets anyway to terminate to `wss`. But `Node-Media-Server` behaves a little bit different, when using Websockets instead of HTTP.

I thought about changing `Node-Media-Server` to have the same behavior as for HTTP and Websockets already. But till now, there was no need for it, because everything worked with Websockets.

**Comment by xqq** [comment]
I think both `http-flv` and `websocket-flv` impementation of `Node-Media-Server` may have the same problem. Analyze your stream pushed from the server, or contact the server developer.

**Comment by baxerus** [comment]
I will try to have a look into this, for sure
